packages/kbn-es-archiver/src/lib/docs/generate_doc_records_stream.ts

```diff
@@ -57,14 +61,25 @@ export function createGenerateDocRecordsStream({
 
           for (const hit of resp.body.hits.hits) {
             remainingHits -= 1;
-            stats.archivedDoc(hit._index);
+
+            if (hasDatastreams && !indexToDatastream.has(hit._index)) {
+              const {
+                [hit._index]: { data_stream: dataStream },
+              } = await client.indices.get({ index: hit._index, filter_path: ['*.data_stream'] });
+              indexToDatastream.set(hit._index, dataStream);
+            }
+
+            const dataStream = indexToDatastream.get(hit._index);
+            stats.archivedDoc(dataStream || hit._index);
+
             this.push({
               type: 'doc',
               value: {
                 // if keepIndexNames is false, rewrite the .kibana_* index to .kibana_1 so that
                 // when it is loaded it can skip migration, if possible
                 index:
                   hit._index.startsWith('.kibana') && !keepIndexNames ? '.kibana_1' : hit._index,
+                data_stream: dataStream,
```

Comment:
Nit: Part of me would prefer that docs either had an index of a data_stream, but I'm not opposed to keeping the index if there's some use for it.

Reply:
I mainly kept it for traceability when debugging or inspecting archived data, besides that there's no real use for it :)